src/scripts/main.js

```diff
@@ -1,5 +1,22 @@
 'use strict';
 
-document.addEventListener('click', e => {
-  // write code here
+const wall = document.querySelector('.wall');
+const spider = document.querySelector('.spider');
+const wallSize = parseInt(window.getComputedStyle(wall).width);
+const spiderSize = parseInt(window.getComputedStyle(spider).width);
+const border = wallSize - (spiderSize / 2);
+
+wall.addEventListener('click', e => {
+  const x = e.offsetX > border
+    ? border
+    : e.offsetX < spiderSize / 2
+      ? spiderSize / 2 : e.offsetX;
```

Comment:
this part of code is very similar to the other one below, create a function for this